Tahi-project/tahi
# APERTA-5828: Add Uploader Component to PRQ card

JIRA issue: https://developer.plos.org/jira/browse/APERTA-5828
#### What this PR does:

Switches the Publishing Related Question attachment uploader to the new standard uploader created in [Aperta-5753](https://github.com/Tahi-project/tahi/pull/1974). This allows an attachment to be uploaded, replaced with a different file, as well as deleted.

Changed relationship between `question_answer` and `question_attachment` so that a `question_answer` can how have multiple `question_attachment`s. We then moved the creation and updating of QuestionAttachments to the `QuestionAttachmentsController`, removing it from the `QuestionAnswersController`. The `nested-question-uploader-manager` now handles the create/update of these attachments rather then setting the answer value.

Also includes setting the attachment caption to the updated, so that the user can set the title of the file that was uploaded (screenshot included below). 
#### How to test
- Pull down branch; run migrations; run `rake nested-questions:seed` if you _did not_ use `rake db:reset`. 
- Run the app ; create a paper; go to the Additional Information card
- Choose "Yes" on the first question (e..g "Have the results.... been published elsewhere?") 
- Upload a file ; after it uploads you should be able to enter a caption. 
- Refresh ; verify your uploaded file still shows up under the question and with your caption
- Upload another file and perhaps another. Then delete one and replace one. 
- If all of those things work then everything is all set and this PR should get a good :white_check_mark:  :)
#### Notes

The standard upload component is also used in the Ethics card ((APERTA-5829)[https://github.com/Tahi-project/tahi/pull/2006]). We had to make minor changes to values passed to components to not affect those changes.

We did not find any other fallout from changing the attachment relationship, and switching the creation/update from the QuestionAnswersController, but a significant enough change to note.
#### Major UI changes

Publishing Related Questions now includes setting of title, as well as multiple file uploads:
![screenshot 2016-01-21 11 27 40](https://cloud.githubusercontent.com/assets/82035/12487316/a7ebc980-c034-11e5-8dd5-12ee0784dafe.png)

---
#### Code Review Tasks:

Author tasks:  

Reviewer tasks:
#### After the Code Review:

Author tasks:
